### DISCUSSION TITLE
Autostart BankId app on smartphone

### DISCUSSION THREAD

**Discussion by FBUtveckling** [ActiveLogin/ActiveLogin.Authentication#457] [Q&A] [answered]
With the last version of Activelogin (9), it is only possible to automatically start the BankId application when it connects from a computer but if one connects from a smartphone, the application does not start automatically and the BankId App's start button is shown. That is the configuration implemented

            services
                .AddBankId(bankId =>
                {
                    bankId.AddDebugEventListener();
                    bankId.UseQrCoderQrCodeGenerator();
                    bankId.UseUaParserDeviceDetection();
                    bankId.UseProductionEnvironment().UseClientCertificate(() => certificates[0]);
                });
            services.AddBankIdSign(bankId =>
            {
                bankId.AddSameDevice(BankIdSignDefaults.SameDeviceConfigKey, "BankID (SameDevice)", options => { });
                bankId.AddOtherDevice(BankIdSignDefaults.OtherDeviceConfigKey, "BankID (OtherDevice)", options => { });
            });
            services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
                .AddBankIdAuth(bankId =>
                {
                    bankId.AddSameDevice(BankIdAuthDefaults.SameDeviceAuthenticationScheme, "BankID (SameDevice)", options => { });
                    bankId.AddOtherDevice(BankIdAuthDefaults.OtherDeviceAuthenticationScheme, "BankID (OtherDevice)", options => { });
                });

I have forgotten something?

**Comment by PeterOrneholm** [comment]
Can you provide more info, like this:
- In what version did it work? 8.0.0?
- The specific version it worked for you and a screenrecording of that flow, then how it works in v9.
- The specific device, browser etc. you are using.
- A repo with a minimal setup of a project where we can reproduce this

No changes were made whatsoever ([from what I can see](https://github.com/ActiveLogin/ActiveLogin.Authentication/compare/v8.0.0...v9.0.0)) between v8 and v9, so my best guess is that it's something else.

**Comment by FBUtveckling** [comment]
- I just updated to version 9, but the previous version didn't work either. It only works to automatically start the BankId application when you work on a computer, but if you work from an Android phone it does not work.
- Samsung android 12 phone.
You already described this problem previously:
https://github.com/ActiveLogin/ActiveLogin.Authentication/issues/251

The problem persists.

Thanks!

**Comment by FBUtveckling** [comment]
It is curious that after 4 years the problem persists.
????

**Reply by PeterOrneholm** [reply, marked as answer]
Well, there have been no reports or issues reported since we closed the PR you are refering to :)
The "issue" here is that Active Login triggers the launch of a third party application "automatically" when the page loads, without any user interaction. Some Android OS versions (probably the later ones) requires user interaction for this because of security. So no change in Active Login, rather more secure versions of Android OS. The `BrowserMightNotAutoLaunch` property indicates just that and shows a button instead.

The solution in Active Login is to restructure some of our fundamental parts so that the button choosing between this/other device does not reload the page but rather opens up the BankID app right away. At the moment this feature isn't prioritized, but if it's important to you, we are more than happy to receive a PR or that you sponsor the feature.
